src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+  const [, setProfileAtom] = useRecoilState(profileAtom);
+
+  const [isLoadingFetchingChannelDetail, setIsLoadingFetchingChannelDetail] = useState(false);
+
+  const __createChannel = () => {
+    PushNotification.createChannel(
+      {
+        channelId: 'bettersosialid', // (required)
+        channelName: 'New Messages & Comments', // (required)
+        playSound: true, // (optional) default: true
+        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function
+        importance: 4, // (optional) default: 4. Int value of the Android notification importance
+        vibrate: true // (optional) default: true. Creates the default vibration patten if true.
+      },
+      (created) => {
+        if (__DEV__) {
+          console.log(`createChannel returned '${created}'`);
+        }
+      } // (optional) callback returns whether the channel was created, false means it already existed.
+    );
+  };
+
+  const __pushNotifIos = (message) => {
+    const {title, body} = message.notification;
+    PushNotificationIOS.addNotificationRequest({
+      title,
+      body,
+      id: message.messageId,
+      userInfo: message.data
+    });
+  };
+
+  const __pushNotifAndroid = (remoteMessage) => {
+    const {title, body} = remoteMessage.notification;
+    PushNotification.localNotification({
+      id: '123',
+      title,
+      channelId: 'bettersosialid',
+      message: body,
+      data: remoteMessage.data
+    });
+  };
+
+  const __handlePushNotif = (remoteMessage) => {
+    const {data} = remoteMessage;
+    if (data.channel_type !== 3) {
+      if (isIos) {
+        __pushNotifIos(remoteMessage);
+      } else {
+        __pushNotifAndroid(remoteMessage);
+      }
+    }
+  };
+
+  const __updateProfileAtomId = async () => {
+    const signedUserId = await getUserId();
+    const anonUserId = await getAnonymousUserId();
+    const token = TokenStorage.get(ITokenEnum.token);
+    const anonymousToken = TokenStorage.get(ITokenEnum.anonymousToken);
+
+    setProfileAtom({
+      anonProfileId: anonUserId,
+      signedProfileId: signedUserId,
+      token,
+      anonymousToken
+    });
+  };
+  let isOpenNotification = false;
+
+  const helperNavigationResetWithData = (screenData) => {
+    setTimeout(() => {
+      if (!isOpenNotification) {
+        isOpenNotification = true;
+        const routes = [
+          {
+            name: 'AuthenticatedStack',
+            params: {
+              screen: 'HomeTabs',
+              params: {
+                screen: 'SignedChannelList'
+              }
+            }
+          }
+        ];
+
+        if (screenData)
+          routes.push({
+            name: 'AuthenticatedStack',
+            params: {
+              ...screenData
+            }
+          });
+
+        navigation.reset({
+          index: screenData ? 2 : 1,
+          routes
+        });
+      }
+    }, 500);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `helperNavigationResetWithData` navigates to a new screen based on the provided data. The use of `setTimeout` here might lead to unexpected behavior if the component unmounts before the timeout finishes. Consider using a cleanup function in a useEffect hook to clear the timeout when the component unmounts.



<!-- This is an auto-generated comment by OSS CodeRabbit -->